_source/_posts/2022-04-26-webauthn-java.md

```diff
@@ -0,0 +1,713 @@
+---
+layout: blog_post
+title: "Building a WebAuthn Application with Java"
+author: john-gramila
+by: contractor
+communities: [java]
+description: "Learn more about the WebAuthn specification and how it can strengthen your application's security. Then build a Java application using WebAuthn for authentication."
+tags: [api, webauthn, java, authentication]
+tweets:
+- "WebAuthn is the future. Are you ready? Find out how to implement it in your work with just a couple of steps, and use Java to build an application that authenticates users in the safest way possible."
+- "Having problems with password abuse? A WebAuthn specification lets developers create passwordless applications with an increasingly broad and accessible range of hardware authenticators. Learn how to build your own WebAuthn application with Java here!"
+image: blog/webauthn-java/java-webauthn-social.jpg
+type: awareness
+---
+
+The [Web Authentication (WebAuthn)](https://webauthn.io/) specification, given [official approval](https://www.w3.org/2019/03/pressrelease-webauthn-rec.html) by the World Wide Web Consortium (W3C) and the FIDO Alliance in 2019, aims to strengthen online security by allowing users to sign in to sites with elements like biometrics and FIDO security keys. The WebAuthn API can replace or supplement less-secure passwords, which may be weak and are often shared.
+
+WebAuthn is supported by default in Firefox and Chrome browsers and can be used in Edge browsers on recent Windows systems. As support for the new standard is built into more devices, more websites will offer this easier, more secure method of authentication.
+
+In this post, I'll explain what WebAuthn is, how it works, and we'll build a Java web application that allows a user to register and log in with a supported WebAuthn device.
+
+{% include toc.md %}
```

Comment:
If I look at this in the [preview](https://deploy-preview-1155--okta-blog.netlify.app/blog/2022/04/26/webauthn-java), I think there are some opportunities for improvement. For example:

- Getting started -> Get started with WebAuthn and Spring Boot
- Data layer -> Data layer overview (or something that explains the section better, maybe add Spring Data JPA)
- An easier way -> An easier way to WebAuthn